modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala

```diff
@@ -86,13 +87,7 @@ class HttpResponseCodeSchemaVisitor
             get: S => AA
         ): ResponseCodeExtractor[S] = {
           val aExt = apply(instance)
-          aExt match {
-            case NoResponseCode => NoResponseCode
-            case RequiredResponseCode(f) =>
-              RequiredResponseCode { s => f(get(s)) }
-            case OptionalResponseCode(f) =>
-              OptionalResponseCode { s => f(get(s)) }
-          }
+          Contravariant[ResponseCodeExtractor].contramap(aExt)(get)
```

Comment:
If this is a contravariant, there must be a type class for the Optional one, right?